compute/disks/disks_test.go

```diff
@@ -184,28 +168,90 @@ func TestComputeDisksSnippets(t *testing.T) {
 			t.Errorf("deleteDisk got %q, want %q", got, want)
 		}
 
+		buf.Reset()
+		want = "Disk created"
+
+		if err := createRegionalDiskFromSnapshot(buf, tc.ProjectID, region, replicaZones, diskName2, diskType, diskSnapshotLink, 50); err != nil {
+			t.Errorf("createRegionalDiskFromSnapshot got err: %v", err)
+		}
+		if got := buf.String(); !strings.Contains(got, want) {
+			t.Errorf("createRegionalDiskFromSnapshot got %q, want %q", got, want)
+		}
+
+		buf.Reset()
+		want = "Disk deleted"
+
+		err = deleteRegionalDisk(buf, tc.ProjectID, region, diskName2)
+		if err != nil {
+			t.Errorf("deleteRegionalDisk got err: %v", err)
+		}
+		if got := buf.String(); !strings.Contains(got, want) {
+			t.Errorf("deleteRegionalDisk got %q, want %q", got, want)
+		}
+
+		if err := deleteDisk(buf, tc.ProjectID, zone, diskName); err != nil {
+			t.Errorf("deleteDisk got err: %v", err)
+		}
+		if got := buf.String(); !strings.Contains(got, want) {
+			t.Errorf("deleteRegionalDisk got %q, want %q", got, want)
+		}
+
+		// clean up
 		err = deleteDiskSnapshot(ctx, tc.ProjectID, snapshotName)
 		if err != nil {
 			t.Errorf("deleteDiskSnapshot got err: %v", err)
 		}
+	})
+
+	t.Run("createEmptyDisk and clone it into a regional disk", func(t *testing.T) {
+		buf.Reset()
+		want := "Disk created"
+
+		if err := createEmptyDisk(buf, tc.ProjectID, zone, diskName, diskType, 20); err != nil {
+			t.Fatalf("createEmptyDisk got err: %v", err)
+		}
+		if got := buf.String(); !strings.Contains(got, want) {
+			t.Errorf("createEmptyDisk got %q, want %q", got, want)
+		}
+
+		if err := createRegionalDiskFromDisk(buf, tc.ProjectID, region, replicaZones, diskName2, diskType, sourceDisk, 30); err != nil {
+			t.Errorf("createRegionalDiskFromDisk got err: %v", err)
+		}
+		if got := buf.String(); !strings.Contains(got, want) {
+			t.Errorf("createRegionalDiskFromDisk got %q, want %q", got, want)
+		}
+
+		// clean up
+		err = deleteRegionalDisk(buf, tc.ProjectID, region, diskName2)
+		if err != nil {
+			t.Errorf("deleteRegionalDisk got err: %v", err)
+		}
 
 		err = deleteDisk(buf, tc.ProjectID, zone, diskName)
 		if err != nil {
 			t.Errorf("deleteDisk got err: %v", err)
 		}
 	})
 
-	t.Run("createEmptyDisk", func(t *testing.T) {
+	t.Run("create, clone and delete an encrypted disk", func(t *testing.T) {
 		buf.Reset()
-		want = "Disk created"
+		want := "Disk created"
 
-		if err := createEmptyDisk(buf, tc.ProjectID, zone, diskName, diskType); err != nil {
-			t.Fatalf("createEmptyDisk got err: %v", err)
+		if err := createEncryptedDisk(buf, tc.ProjectID, zone, diskName, diskType, 20, "SGVsbG8gZnJvbSBHb29nbGUgQ2xvdWQgUGxhdGZvcm0=", "", "", ""); err != nil {
```

Comment:
This set up step can be moved into `TestMain(m *testing.M)`.

Reply:
This is not setup - this is the test body! It is calling a code sample wrapped into a function and ensuring that it works.